strings/longest_substr_without_repeating_chars.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Greedy algorithm to find length of longest substring without repeating
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂